scripts/scr_planet_map/scr_planet_map.gml

```diff
@@ -0,0 +1,543 @@
+/* 
+    This function generates a X by Y grid for a planet with all the tile information .
+        In here 2 basic structures are created: the hexGrid and the tile_info
+        Hex grid is just to have the basic hex map
+        All the information regarding the tile and if it has buildings or any other feature is in the tile_info. 
+            The names are pretty self explenatory in regards to what each does
+*/
+function scr_planet_map(planet_type,grid_width, grid_height){
+
+    var hexGrid = ds_grid_create(grid_width, grid_height);
+    var tile_info = ds_grid_create(grid_width, grid_height);
+    var settlement_count = 0;
+
+    for (var x = 0; x < grid_width; x++) {
+        for (var y = 0; y < grid_height; y++) {
+            // Calculate the position of each hexagon
+            var hex_size = 32; // Size of each hexagon
+            var hex_width = hex_size * sqrt(3); // Width of a hexagon
+            var hex_height = hex_size * 2; // Height of a hexagon
+            var xpos = x * (hex_width * 0.75);
+            var ypos = y * (hex_height + (hex_height / 2)) - (x % 2) * (hex_height / 2);
+
+            var monastery_exists = false;
+
+            // Store the hexagon's position and planet type in the hex grid
+            ds_grid_set(hexGrid, x, y, [xpos, ypos, planet_type]);
+            // TODO continue with the definition of the data structure
+            // Initialize tile_info data structure 
+            tile_info[#movement_cost_land, x, y] = 1;
```

Comment:
if tile info is a ds_grid, how are you accessing three dimensions? (this might just be my ignorance)